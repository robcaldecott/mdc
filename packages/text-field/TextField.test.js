import React from "react";
import { TextField, TextFieldHelperText, TextFieldIcon } from ".";
import renderer from "react-test-renderer";

it("renders", () => {
  expect(renderer.create(<TextField />)).toMatchSnapshot();
});

it("renders with a label", () => {
  expect(renderer.create(<TextField label="Label" />)).toMatchSnapshot();
});

it("renders disabled", () => {
  expect(
    renderer.create(<TextField label="Label" disabled />)
  ).toMatchSnapshot();
});

it("renders full width", () => {
  expect(
    renderer.create(<TextField label="Label" fullWidth />)
  ).toMatchSnapshot();
});

it("renders dense", () => {
  expect(renderer.create(<TextField label="Label" dense />)).toMatchSnapshot();
});

it("renders with a box", () => {
  expect(renderer.create(<TextField label="Label" box />)).toMatchSnapshot();
});

it("renders with helper text", () => {
  expect(
    renderer.create(
      <TextField label="Label" helperText={<TextFieldHelperText />} />
    )
  ).toMatchSnapshot();
});

it("renders with persistent helper text", () => {
  expect(
    renderer.create(
      <TextField
        label="Label"
        helperText={<TextFieldHelperText persistent />}
      />
    )
  ).toMatchSnapshot();
});

it("renders with validation helper text", () => {
  expect(
    renderer.create(
      <TextField
        label="Label"
        helperText={<TextFieldHelperText validation />}
      />
    )
  ).toMatchSnapshot();
});

it("renders with a leading icon", () => {
  expect(
    renderer.create(
      <TextField
        label="Label"
        box
        leadingIcon={<TextFieldIcon>event</TextFieldIcon>}
      />
    )
  ).toMatchSnapshot();
});

it("renders with a trailing icon", () => {
  expect(
    renderer.create(
      <TextField
        label="Label"
        box
        trailingIcon={<TextFieldIcon>delete</TextFieldIcon>}
      />
    )
  ).toMatchSnapshot();
});
