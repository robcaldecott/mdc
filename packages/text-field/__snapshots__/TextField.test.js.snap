// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded"
>
  <input
    className="mdc-text-field__input"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  />
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders dense 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded mdc-text-field--dense"
>
  <input
    className="mdc-text-field__input"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  >
    Label
  </span>
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders disabled 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded mdc-text-field--disabled"
>
  <input
    className="mdc-text-field__input"
    disabled={true}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  >
    Label
  </span>
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders full width 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded mdc-text-field--fullwidth"
>
  <input
    className="mdc-text-field__input"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  >
    Label
  </span>
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders with a box 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded mdc-text-field--box"
>
  <input
    className="mdc-text-field__input"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  >
    Label
  </span>
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders with a label 1`] = `
<label
  className="mdc-text-field mdc-text-field--upgraded"
>
  <input
    className="mdc-text-field__input"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    value={undefined}
  />
  <span
    className="mdc-text-field__label"
  >
    Label
  </span>
  <div
    className="mdc-text-field__bottom-line"
  />
</label>
`;

exports[`renders with helper text 1`] = `
Array [
  <label
    className="mdc-text-field mdc-text-field--upgraded"
  >
    <input
      className="mdc-text-field__input"
      disabled={undefined}
      onBlur={[Function]}
      onFocus={[Function]}
      value={undefined}
    />
    <span
      className="mdc-text-field__label"
    >
      Label
    </span>
    <div
      className="mdc-text-field__bottom-line"
    />
  </label>,
  <p
    className="mdc-text-field-helper-text"
  />,
]
`;

exports[`renders with persistent helper text 1`] = `
Array [
  <label
    className="mdc-text-field mdc-text-field--upgraded"
  >
    <input
      className="mdc-text-field__input"
      disabled={undefined}
      onBlur={[Function]}
      onFocus={[Function]}
      value={undefined}
    />
    <span
      className="mdc-text-field__label"
    >
      Label
    </span>
    <div
      className="mdc-text-field__bottom-line"
    />
  </label>,
  <p
    className="mdc-text-field-helper-text mdc-text-field-helper-text--persistent"
  />,
]
`;

exports[`renders with validation helper text 1`] = `
Array [
  <label
    className="mdc-text-field mdc-text-field--upgraded"
  >
    <input
      className="mdc-text-field__input"
      disabled={undefined}
      onBlur={[Function]}
      onFocus={[Function]}
      value={undefined}
    />
    <span
      className="mdc-text-field__label"
    >
      Label
    </span>
    <div
      className="mdc-text-field__bottom-line"
    />
  </label>,
  <p
    className="mdc-text-field-helper-text mdc-text-field-helper-text--validation-msg"
  />,
]
`;
